tofu/geom/_core_optics.py

```diff
@@ -1985,6 +2077,11 @@ def plot_line_on_det_tracing(
         bragg = self._checkformat_bragglamb(lamb=lamb, n=n)
         if nphi is None:
             nphi = 100
+        nphi2 = (nphi/2.)
+        if (nphi2 % 2) == 0.:
+            nphi2 = int(nphi2 - 1)
+        else:
+            nphi2 = int(nphi2 - 0.5)
```

Comment:
Simpler to do it the other way around: define `nphi2`, then build `nphi` to be odd or even depending on what you need:

```
if nphi2 is None:
    nphi2 = 50

nphi = 2*nphi2 
# or
nphi = 2*nphi2 + 1
```